sympy/sets/sets.py

```diff
@@ -1968,6 +1968,14 @@ def __lt__(self, other):
             raise TypeError("Invalid comparison of set with %s" % func_name(other))
         return self.is_proper_subset(other)
 
+    def __eq__(self, other):
```

Comment:
It seems the `__eq__` you are trying to write here is the same as `Basic.__eq__`, so I don't think it should be needed. I would try to figure out why this stopped working.